mdx/guides/build-your-first-cfl-enabled-smart-contract.mdx

```diff
@@ -0,0 +1,270 @@
+
+<Notification type="alert">Use a test account, DO NOT use an account with your personal savings.</Notification>
+
+Now that you have setup the `.env` file, let’s install the npm packages:
+
+```shell
+$ npm install
+```
+
+We will be interacting mainly with the `examples/` folder’s .ts files; they are scripts that represent web3 interactions with your smart contract that one may find in a Dapp’s front end. For this guide, we will be mainly concerned with the web3/simple_CFL_token_swap.ts file:
+
+```javascript
+// libraries
+import * as qs from 'qs';
+import * as fetch from 'node-fetch';
+
+// utils
+import { baseUnitAmount, setUpWeb3GanacheAsync, fetchERC20BalanceFactory } from './utils';
+import { migrationAsync } from '../migrations/migration';
+
+// wrappers
+// TODO: wrapper imports goes here
+
+// constants
+const ETHEREUM_RPC_URL = process.env.ETHEREUM_RPC_URL;
+const MNEMONIC = process.env.MNEMONIC;
+const DAI_CONTRACT = '0x6b175474e89094c44da98b954eedeac495271d0f'; // DAI mainnet contract address
+
+(async () => {
+    // initialize ganache fork and deploy contracts
+    const { web3Wrapper, provider } = await setUpWeb3GanacheAsync(MNEMONIC, ETHEREUM_RPC_URL);
+    const { simpleTokenSwapAddress } = await migrationAsync(provider, web3Wrapper);
+    
+    // handy util to check address balance of DAI
+    const fetchDAIBalanceAsync = fetchERC20BalanceFactory(provider, DAI_CONTRACT);
+
+    // 1. call 0x api for a quote for one dollar of DAI.
+
+    // TODO: write fetch GET call to get a swap quote from 0x API
+
+    // 2. send response from 0x api to your smart contract
+
+    // TODO: write web3 smart contract interaction
+
+})()
+```
+
+## Fetching a quote from 0x API 🥏
+
+As described in the [Overview section](/docs/guides/build-your-first-cfl-enabled-smart-contract#overview), 0x-powered CFL begins off-chain, before any web3 interactions are made. For this guide, we will be buying 1 DAI (specified by a variable in the script) with ETH (note, not WETH). By specifying `buyAmount` instead of `sellAmount`, the API understands you intend to buy 1 unit of DAI, not sell 1 unit of ETH.
+
+```javascript
+    (async () => {
+        ...
+        // 1. call 0x API for a quote for one dollar of DAI.
+        const buyAmount = baseUnitAmount(1);
+
+        const params = {
+            sellToken: 'ETH',
+            buyToken: 'DAI',
+            buyAmount: buyAmount.toString(),
+        }
+
+        const res = await fetch(`https://api.0x.org/swap/quote?${qs.stringify(params)}`);
+        const quote = await res.json();
+
+        console.log('Received quote:', quote);
+        ...
+    })()
+```
+
+*Note: baseUnitAmount is a handy util tool provided that adjusts a number by the number of decimals an asset has. In DAI’s case, there is 18 decimals; baseUnitAmount would be buying 1 DAI.*
```

Comment:
I would move away from this "Note:*" pattern. For short warnings, it might be cool to use the little notification islands that we have available, or I'd just work it into the main text.